example/src/App.tsx

```diff
@@ -156,25 +151,33 @@ const App = () => {
 };
 
 function RenderResults() {
-  const groups = useMatches();
-  const flattened = React.useMemo(
-    () =>
-      groups.reduce((acc, curr) => {
-        acc.push(curr.name);
-        acc.push(...curr.actions);
-        return acc;
-      }, []),
-    [groups]
-  );
+  const { throttledGroups: matches, throttledRootActionId: rootActionId } =
+    useDeepMatches();
+
+  const flattened = React.useMemo(() => {
+    let res = [];
+    for (let i = 0; i < matches.length; i++) {
+      let match = matches[i];
+      if (match !== NO_GROUP) res.push(match.name);
+      for (let j = 0; j < match.actions.length; j++) {
+        res.push(match.actions[j]);
+      }
+    }
+    return res;
```

Comment:
Average run times with 100k results:
 
reduce: .75ms
flatMap: 9ms
for loop: .5ms

Reply:
for loops fo life

Reply:
Idea is for no one to have 100k actions but who knows, go crazy

Reply:
I still think this flattening should be contained within the lib :P

Reply:
Makes sense, `useDeepMatches` can return the flattened results as users have to flatten the results regardless